index.js

```diff
@@ -4,65 +4,107 @@ console.log(fifaData);
-
-function getWinners(/* code here */) {
-
-    /* code here */
-
+console.log(getYears(getFinals(fifaData)));
+
+/* Task 4: Implement a higher-order function called `getWinners`, that accepts the callback function `getFinals()` and determine the winner (home or away) of each `finals` game. Return the name of all winning countries in an array called `winners` */
+
+function getWinners(getFinals) {
+    let winners = [];
+    getFinals.forEach(function(element) {
+        let winner = '';
+        let homeScore = (element['Home Team Goals']) + (element['Half-time Home Goals']);
+        let awayScore = (element['Away Team Goals']) + (element['Half-time Away Goals']);
+        if (homeScore > awayScore) {
+            winner = element['Home Team Name'];
+        } else if (homeScore < awayScore) {
+            winner = element['Away Team Name'];
+        } else {
+            // This is part of the stretch goal listed in the readMe
+            winner = element['Win conditions'].split(' win ')[0];
+        }
+        winners.push(winner);
+    });
+    return winners;
 };
 
-getWinners();
+console.log(getWinners(getFinals(fifaData)));
 
 /* Task 5: Implement a higher-order function called `getWinnersByYear` that accepts the following parameters and returns a set of strings "In {year}, {country} won the world cup!" 
 
 Parameters: 
  * callback function getWinners
  * callback function getYears
  */
-
-function getWinnersByYear(/* code here */) {
-
+function getWinnersByYear(getWinners, getYears) {
+    let string = ``;
+    getWinners.forEach(function(item, index) {
+        string += `In ${getYears[index]}, ${item} won the world cup!\n`;
+    });
+    return string;
 };
 
-getWinnersByYear();
+console.log(getWinnersByYear(
+    (getWinners(getFinals(fifaData))), (getYears(getFinals(fifaData)))
+));
 
 /* Task 6: Write a function called `getAverageGoals` that accepts a parameter `data` and returns the the average number of home team goals and away team goals scored per match (Hint: use .reduce and do this in 2 steps) */
 
-function getAverageGoals(/* code here */) {
+function getAverageGoals(data) {
+    let homeScores = data.map(function(item) {
+        return item['Home Team Goals'] + item['Half-time Home Goals'];
+    });
+    let homeTotal = homeScores.reduce(((acc, item) => acc + item), 0);
+    let homeAvg = homeTotal / data.length;
 
-    /* code here */
+    let awayScores = data.map(function(item) {
+        return item['Away Team Goals'] + item['Half-time Away Goals'];
+    });
+    let awayTotal = awayScores.reduce(((acc, item) => acc + item), 0);
+    let awayAvg = awayTotal / data.length;
 
+    return `Average number of home team goals: ${homeAvg}\nAverage number of away team goals: ${awayAvg}`;
```

Comment:
The code looks great! But to avoid the crazy floating-point numbers, you could implement `${Math.round(homeAvg)}` on each individual template literal to get whole numbers. 🙂 
`Average number of home team goals: 2.5217391304347827` vs `Average number of home team goals: 3`

Reply:
That makes sense! Got it. Will keep that in mind next time